src/nvim/eval.c

```diff
@@ -17685,6 +17685,15 @@ static void f_writefile(typval_T *argvars, typval_T *rettv, FunPtr fptr)
     EMSG2(_(e_listarg), "writefile()");
     return;
   }
+  const list_T *const list = argvars[0].vval.v_list;
+  if (list == NULL) {
+    return;
+  }
+  TV_LIST_ITER_CONST(list, li, {
+    if (tv_get_string_chk(TV_LIST_ITEM_TV(li)) == NULL) {
```

Comment:
There is `tv_check_str_or_nr()` function now, it should be used here in place of `tv_get_string_chk()` which will do useless job of formatting number.